Trust-Login-C2C-(Source) (#4835)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/trust-login-source.md

---
id: trust-login-source
title: Trust Login Source

> sidebar_label: Trust Login

Reviewer: I think all instances of "Trust Login" might need to be changed to "TrustLogin". Check with the product manager or development lead.

Author: @jpipkin1 confirmed with @draval-sumo . The correct naming for the integration is "Trust Login".

description: This integration aims to collect the Report Logs from the Trust Login and send them to Sumo Logic.
---

import CodeBlock from '@theme/CodeBlock';
import ExampleJSON from '/files/c2c/trust-login/example.json';
import MyComponentSource from '!!raw-loader!/files/c2c/trust-login/example.json';
import TerraformExample from '!!raw-loader!/files/c2c/trust-login/example.tf';
import ForwardToSiem from '/docs/reuse/forward-to-siem.md';
import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/send-data/trust-login-icon.png')} alt="trust-login-icon" width="50" />

> Trust Login is an advanced authentication platform that offers a secure and seamless login solutions to users and organizations alike. Trust Login enhances the authentication process by prioritizing trust, ease of use, and strong security. This ensures reliable access control without compromising user experience.
> 
> The purpose of this integration is to provide Sumo Logic the Report Logs that are gathered from the Trust Login.

Reviewer: ```suggestion
Trust Login is an advanced authentication platform that offers a secure and seamless login solution to users and organizations alike. Trust Login enhances the authentication process by prioritizing trust, ease of use, and strong security. This ensures reliable access control without compromising user experience.

The purpose of this integration is to provide Sumo Logic the Report Logs that are gathered from Trust Login.
```

:::important
The Trust Login API documentation is not public and can only be accessed by partners or customers.
:::

## Data collected

| Polling Interval | Data |
| :--- | :--- |
| 5 min | Report logs |

## Setup

### Vendor configuration

> The Trust Login Source requires authentication with a bearer token associated with ApiToken.

Reviewer: ```suggestion
The Trust Login Source requires authentication with a bearer token associated with an API Token.
```

### Source configuration

> When you create a Trust Login Source, you add it to a Hosted Collector. Before creating the Source, identify the Hosted Collector you want to use or create a new Hosted Collector. For instructions, see [Configure a Hosted Collector](/docs/send-data/hosted-collectors/configure-hosted-collector).

Reviewer: ```suggestion
When you create a Trust Login Source, you add it to a Hosted Collector. Before creating the Source, identify the Hosted Collector you want to use or create a new Hosted Collector. For instructions, see [Configure a Hosted Collector and Source](/docs/send-data/hosted-collectors/configure-hosted-collector).
```

1. (Optional) **Fields.** Click the **+Add Field** link to define the fields you want to associate, each field needs a name (key) and value.
   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo that does not exist in the Fields schema it is ignored, known as dropped. 
1. **Base URL**. Provide the bearer token collected from the Trust Login platform.
1. The **Polling Interval** is set for five minutes by default, you can adjust it based on your needs.
1. **Processing Rules**. Configure any desired filters, such as allowlist, denylist, hash, or mask, as described in [Create a Processing Rule](/docs/send-data/collection/processing-rules/create-processing-rule).
1. When you are finished configuring the Source, click **Save**.

## JSON schema

> Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [how to use JSON to configure Sources](/docs/send-data/use-json-configure-sources) for details. 

Reviewer: ```suggestion
Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [Use JSON to Configure Sources](/docs/send-data/use-json-configure-sources) for details. 
```
